dnf/plugin.py

```diff
@@ -202,30 +202,34 @@ def _iter_py_files(paths, skips, enable_plugins):
     plugins = []
     skips = set(skips)
     enable_plugins = set(enable_plugins)
-    pattern_found = set()
+    pattern_enable_found = set()
+    pattern_disable_found = set()
     for p in paths:
         for fn in glob.glob('%s/*.py' % p):
             (name, base_path) = os.path.splitext(os.path.basename(fn))
             matched = True
             enable_pattern_tested = False
             for pattern_skip in skips:
                 if fnmatch.fnmatch(name, pattern_skip):
-                    pattern_found.add(pattern_skip)
+                    pattern_disable_found.add(pattern_skip)
                     matched = False
                     for pattern_enable in enable_plugins:
                         if fnmatch.fnmatch(name, pattern_enable):
                             matched = True
-                            pattern_found.add(pattern_enable)
+                            pattern_enable_found.add(pattern_enable)
                     enable_pattern_tested = True
             if not enable_pattern_tested:
                 for pattern_enable in enable_plugins:
                     if fnmatch.fnmatch(name, pattern_enable):
-                        pattern_found.add(pattern_enable)
+                        pattern_enable_found.add(pattern_enable)
             if matched:
                 plugins.append(fn)
-    not_found = skips.difference(pattern_found).union(enable_plugins.difference(pattern_found))
-    for pattern in sorted(not_found):
-        logger.warning(_("No plugin match for: {}").format(pattern))
+    enable_not_found = enable_plugins.difference(pattern_enable_found)
+    logger.warning(_("No plugin match for patterns to plugin enable: {}").format(
+        ", ".join(sorted(enable_not_found))))
+    disable_not_found =skips.difference(pattern_disable_found)
+    logger.warning(_("No plugin match for patterns to plugin disable: {}").format(
+        ", ".join(sorted(disable_not_found))))
```

Comment:
Thanks for splitting the message into two, however, the formulation you've used isn't great... Not entirely sure mine is great either, but how about:

_No matches found for the following enable plugin patterns:_

Also, does this not always print the messages, even if all patterns matched?